src/ngraph/op/util/broadcast_base.cpp

```diff
@@ -267,3 +269,75 @@ void op::util::BroadcastBase::generate_adjoints(autodiff::Adjoints& adjoints,
         throw ngraph_error("Autodiff not supported on dynamic op variants");
     }
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    inline bool
+        evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out, const AxisSet& broadcast_axes)
+    {
+        using T = typename element_type_traits<ET>::value_type;
+        runtime::reference::broadcast<T>((arg0->get_data_ptr<ET>()),
+                                         (out->get_data_ptr<ET>()),
+                                         arg0->get_shape(),
+                                         out->get_shape(),
+                                         broadcast_axes);
+        return true;
+    }
+
+    bool evaluate_broadcast(const HostTensorPtr& arg0,
+                            const HostTensorPtr& out,
+                            const std::pair<bool, AxisSet> pair_broadcast_axes,
+                            const Shape output_shape)
+    {
+        if (arg0->get_element_type() != out->get_element_type())
+        {
+            return false;
+        }
+        if (!pair_broadcast_axes.first)
+        {
+            // broadcast_axes not known deterministically
+            return false;
+        }
+        bool rc = true;
+        Shape in_shape = arg0->get_shape();
+        out->set_shape(output_shape);
+        switch (arg0->get_element_type())
+        {
+            TYPE_CASE(boolean)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(i8)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(i16)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(i32)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(i64)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(u8)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(u16)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(u32)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(u64)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(bf16)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(f16)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(f32)(arg0, out, pair_broadcast_axes.second);
+            break;
+            TYPE_CASE(f64)(arg0, out, pair_broadcast_axes.second);
+            break;
+        default: rc = false; break;
+        }
+        return rc;
+    }
+}
+
+bool op::util::BroadcastBase::evaluate(const HostTensorVector& outputs,
+                                       const HostTensorVector& inputs)
+{
+    return evaluate_broadcast(inputs[0], outputs[0], get_broadcast_axes(), get_output_shape(0));
```

Comment:
`get_output_shape(0)` only works for static shapes. you need to look at arg1 and then do the shape merge with the arg0 shape arg2 axes value and the broadcast mode to compute the output shape and use that,

Reply:
An additional complication is that there is a 2-input and a 3-input form of broadcast